runtime/dancebox/src/lib.rs

```diff
@@ -704,13 +726,90 @@ impl pallet_authority_assignment::Config for Runtime {
     type AuthorityId = NimbusId;
 }
 
+pub struct ChargeForBlockCredit<Runtime>(PhantomData<Runtime>);
+impl OnChargeForBlockCredit<Runtime> for ChargeForBlockCredit<Runtime> {
+    fn charge_credits(
+        payer: &AccountId,
+        _para_id: &ParaId,
+        _credits: u32,
+        fee: u128,
+    ) -> Result<(), pallet_services_payment::Error<Runtime>> {
+        use frame_support::traits::tokens::imbalance::Imbalance;
+
+        let result = <Balances as Currency<AccountId>>::withdraw(
+            payer,
+            fee,
+            WithdrawReasons::FEE,
+            ExistenceRequirement::AllowDeath,
+        );
+        let imbalance = result
+            .map_err(|_| pallet_services_payment::Error::InsufficientFundsToPurchaseCredits)?;
+
+        if imbalance.peek() != fee {
+            panic!("withdrawn balance incorrect");
+        }
+
+        Ok(())
+    }
+}
+
+pub const FIXED_BLOCK_PRODUCTION_COST: u128 = 100;
```

Comment:
this means we will charge 100 tokens

Reply:
0.000...100 actually

Reply:
haha I meant 100 units sorry, I will come back with a value for this